lib/plan-finder.js

```diff
@@ -259,10 +279,47 @@ class PlanFinder {
         if (!this.isMimeType(outputInstructions.type)) {
             throw new RenditionFormatUnsupportedError(`outputInstructions.type (${outputInstructions.type}) is not propertly formatted`);
         }
+
         const plan = this.BFS(source, outputInstructions);
         return this.formPlan(source, outputInstructions, plan);
     }
 
+    async getImageMetadata(path) {
+        let metadata = {};
+        if (path && (await fs.pathExists(path))) {
+            const command = `exiftool -json "${path}"`;
+
+            try {
+                console.log(`Running command: ${command}`);
+                const stdout = child_process.execSync(command, {
+                    timeout: EXIFTOOL_TIMEOUT_MS
+                });
+                const rawJSON = JSON.parse(stdout)[0];
+                metadata = rawJSON;
+            } catch (error) {
+                console.error(`error getting metadata via exiftool for source file: ${path}`);
+                console.error(error);
+            }
+
+        } else if (typeof (path) === 'string' && validUrl.isHttpsUri(path)) {
+            try {
+                const command = `curl -s "${path}" | exiftool -json - `;
+                console.log(`Running command: ${command}`);
+                const stdout = child_process.execSync(command, {
```

Comment:
Why use `execSync`? The function is already async, you could use the async version of the exec call.